indico/queries/export.py

```diff
@@ -0,0 +1,214 @@
+from indico.errors import IndicoNotFound, IndicoRequestError
+from indico.types.export import Export
+from indico.queries.storage import RetrieveStorageObject
+
+
+class _CreateExport(GraphQLRequest):
+    query = """
+        mutation CreateExport(
+            $datasetId: Int!, 
+            $columnIds: [Int], 
+            $subsetIds: [Int], 
+            $labelsetIds: [Int],
+            $fileInfo: Boolean,
+            $combineLabels: Boolean,
+            $anonymous: Boolean
+        ) {
+            createExport(
+                datasetId: $datasetId,
+                columnIds: $columnIds,
+                subsetIds: $subsetIds,
+                fileInfo: $fileInfo,
+                labelsetIds: $labelsetIds,
+                combineLabels: $combineLabels
+                anonymous: $anonymous
+            ) {
+                id,
+                datasetId,
+                name,
+                status,
+                downloadUrl,
+                columnIds,
+                labelsetIds,
+                subsetIds
+            }
+        }    
+
+    """
+
+    def __init__(
+        self,
+        dataset_id: int,
+        column_ids: List[int] = None,
+        subset_ids: List[int] = None,
+        labelset_ids: List[int] = None,
+        file_info: bool = None,
+        combine_labels: bool = None,
+        anonymoous: bool = None,
+    ):
+        super().__init__(
+            self.query,
+            variables={
+                "datasetId": dataset_id,
+                "columnIds": column_ids,
+                "subsetIds": subset_ids,
+                "labelsetIds": labelset_ids,
+                "fileInfo": file_info,
+                "combineLabels": combine_labels,
+                "anonymous": anonymoous,
+            },
+        )
+
+    def process_response(self, response):
+        response = super().process_response(response)
+        return Export(**response["createExport"])
+
+
+class GetExport(GraphQLRequest):
+    """
+    Get information on an Export job
+
+    Args:
+        export_id (int): Id of an Export
+
+    Returns:
+        Export object
+
+    """
+
+    query = """
+        query GetExportStatus($exportIds: [Int]) {
+            exports (exportIds: $exportIds) {
+                exports {
+                id
+                datasetId
+                name 
+                status
+                columnIds
+                labelsetIds
+                subsetIds
+                numLabels
+                anonymous
+                downloadUrl
+                createdAt
+                }
+            }
+        }
+    """
+
+    def __init__(self, export_id: int):
+        super().__init__(self.query, variables={"exportIds": [export_id]})
+
+    def process_response(self, response):
+        response = super().process_response(response)
+        return Export(**response["exports"]["exports"][0])
+
+
+class RetrieveExport(RetrieveStorageObject):
```

Comment:
Should this be _RetrieveExport? Do we expect people to call this directly?

Reply:
good point. updated to RetrieveExport. DownloadExport is the public query